### PR TITLE
feat(index-bar): support anchor name slot

### PR DESCRIPTION
**Pull request by YHoney7**
## Checklist

List of tasks you have already done and plan to do.

- [ ] Fix linting errors
- [ ] Tests have been added / updated (or snapshots)

## Change information

Describe your modifications here.

## Issues

The issues you want to close, formatted as close #1.

## Related Links

Links related to this pr.

### CODE REVIEW

File: packages/varlet-ui/types/indexBar.d.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export class IndexBar extends VarComponent {
 
   $slots: {
     default(): VNode[]
+    'anchor-name': VNode[]
```
Comment (chouchouji) [inline packages/varlet-ui/types/indexBar.d.ts:32]: 上面加个换行

Comment (chouchouji) [inline packages/varlet-ui/types/indexBar.d.ts:32]: 没加参数
image

File: packages/varlet-ui/types/indexBar.d.ts [hunk 2]
```diff
@@ -22,13 +22,19 @@ export interface IndexBarProps extends BasicAttributes {
   cssMode?: boolean
 }
 
+export interface IndexBarIndicatorData {
```
Comment (chouchouji) [inline packages/varlet-ui/types/indexBar.d.ts:25]: 这个就不要了

File: packages/varlet-ui/types/indexBar.d.ts [hunk 3]
```diff
@@ -22,13 +22,19 @@ export interface IndexBarProps extends BasicAttributes {
...
+}
+
 export class IndexBar extends VarComponent {
   static setPropsDefaults: SetPropsDefaults<IndexBarProps>
 
   $props: IndexBarProps
 
   $slots: {
     default(): VNode[]
+
```
Comment (chouchouji) [inline packages/varlet-ui/types/indexBar.d.ts:36]: 这个换行，我的锅，之前没看清楚，项目里面的其他都不要这个换行的

**Review by chouchouji — APPROVED** [review]

**Review by BeADre — APPROVED** [review]

**Pull request merged**